pkg/apis/operator/v1/types.go

```diff
@@ -369,18 +369,67 @@ func (cp CNIPluginType) String() string {
 	return string(cp)
 }
 
+type IPAMPluginType string
+
+const (
+	IPAMPluginCalico    IPAMPluginType = "Calico"
+	IPAMPluginHostLocal IPAMPluginType = "HostLocal"
+	IPAMPluginAmazonVPC IPAMPluginType = "AmazonVPC"
+	IPAMPluginAzureVNET IPAMPluginType = "AzureVNET"
+)
+
+var IPAMPluginTypes []IPAMPluginType = []IPAMPluginType{
+	IPAMPluginCalico,
+	IPAMPluginHostLocal,
+	IPAMPluginAmazonVPC,
+	IPAMPluginAzureVNET,
+}
+
+var IPAMPluginTypesString []string = []string{
+	IPAMPluginCalico.String(),
+	IPAMPluginHostLocal.String(),
+	IPAMPluginAmazonVPC.String(),
+	IPAMPluginAzureVNET.String(),
+}
+
+func (cp IPAMPluginType) String() string {
+	return string(cp)
+}
+
+type IPAMSpec struct {
+	// Specifies the IPAM plugin that will be used in the Calico or Calico Enterprise installation.
+	// * For CNI Plugin Calico, this field defaults to Calico.
```

Comment:
This generates the following
```
<td>
<code>type</code><br>
<em>
<a href="#operator.tigera.io/v1.IPAMPluginType">
IPAMPluginType
</a>
</em>
</td>
<td>
<p>Specifies the IPAM plugin that will be used in the Calico or Calico Enterprise installation.</p>
<ul>
<li><p>For CNI Plugin Calico, this field defaults to Calico.</p></li>
<li><p>For CNI Plugin GKE, this field defaults to HostLocal.</p></li>
<li><p>For CNI Plugin AzureVNET, this field defaults to AzureVNET.</p></li>
<li><p>For CNI Plugin AmazonVPC, this field defaults to AmazonVPC.</p></li>
</ul>
<p>The IPAM plugin is installed and configured only if the CNI plugin is set to Calico,
for all other values of the CNI plugin the plugin binaries and CNI config is a dependency
that is expected to be installed separately.</p>
<p>Default: Calico</p>
</td>
```
So it does what I was hoping though the CRD description looks a little funky.